your-toolbelt.md

```diff
@@ -0,0 +1,47 @@
+### Your toolbelt
+
+
+#### Exercise: Uses and understands unit, functional and integration tests
+
+1. You've now written two types of tests. Name them, and explain the differences.
+The test's writen were the Unit Tests and the Integration Tests. The Unit Tests are to teste a small piece of code, mainly to test the models (the views and the controllers aren't so much tested, in case of the views because it can change in a short period of time) Ideally, there will be, at least, one test for each method. The Integration Tests are maded to test differentes pieces of code of the system and garantee the they work together. In the simple sinatra sitiuation, the integration tests were to navegate in the routes, fill a specific field, click a button and expect the page to have a specific text.
```

Comment:
Watch out when you say that unit tests are testing models, that is not entirely true. You can have unit tests that test models, but unit tests test unit. A unit can be a lot of this, but for the sake of simplicity let's assume for here on out that when we refer to a unit we are talking about a class. 

That class can be any class, what we commonly call a model in the rails world, which is a class that inherits from `ActiveRecord::Base`, or a Plain Old Ruby Object (PORO).

A PORO is any class that you are already used to. So, if you have this:

```ruby
class X
  def sum(a, b)
    a + b
  end
end
```

And the matching test:

```ruby
describe X do
  it 'sums two numbers' do
    expect(X.new.sum(2, 2)).to eq 4
  end
end
```

This is a unit test. The difference to an integration test is that an integration test will test multiple units and how they interact with each other, usually from the point of view of the "user". In the case of the web, that is usually by interacting with a web page through a browser, be it headless or not.